### PR TITLE
Preview release of 0.2.0

### PR DESCRIPTION
**Pull request by adrianbrink**
This PR request updates ethermint to its newest dependencies (go-ethereum and tendermint libs).

Once this lands, the focus will be on feature addition and bug fixes to get ethermint to be production ready.

### CODE REVIEW

**Comment by adrianbrink** [comment]
@ebuchman

File: README.md [hunk 1]
```diff
@@ -2,51 +2,68 @@
...
-### Run
+#### Geth
+
+Lastly, you can use all the cool ethereum tools that you are already used to. For example, you can use geth to interact with the ethereum instance.
 
-Now just run `ethermint --datadir ~/.ethermint`.  You should see Tendermint blocks start streaming by.
-In another window, run `geth attach ~/.ethermint/geth.ipc` to drop into a web3 console for
-your ethermint node:
+```
+get attach http://localhost:8545
```
Comment (illya13) [inline README.md]: `geth` ?

**Comment by illya13** [comment]
Do we have CI build ?

**Comment by adrianbrink** [comment]
@illya13 Yeah, that passed :-)

File: README.md (hunk 1 above)
Comment (adrianbrink) [reply README.md]: So, that people can interact with the ethereum node

Comment (illya13) [reply README.md]: You did not get me. There is a typo above. It should be `geth attach...` but we have `get attach`

Comment (adrianbrink) [reply README.md]: oh right, fixed now :-)

File: ethereum/backend.go [hunk 2]
```diff
@@ -39,17 +39,11 @@ func NewBackend(ctx *node.ServiceContext, config *eth.Config, client rpcClient.H
...
-	// TODO: Can we build it ourselves so it doesnt need to come from registering
-	// this as a Service on the node ?
 	ethereum, err := eth.New(ctx, config, p)
 	if err != nil {
 		return nil, err
 	}
 
-	// send special event to go-ethereum to switch homestead=true
-	// currentBlock := ethereum.BlockChain().CurrentBlock()
-	// ethereum.EventMux().Post(core.ChainHeadEvent{currentBlock})
```
Comment (ramilexe) [inline ethereum/backend.go]: Why did you removed it? It is fix for https://github.com/tendermint/ethermint/issues/14

File: ethereum/pending.go [hunk 3]
```diff
@@ -160,9 +160,8 @@ func (w *work) deliverTx(blockchain *core.BlockChain, config *eth.Config, chainC
 
 	logs := w.state.GetLogs(tx.Hash())
 
-	w.txIndex += 1
+	w.txIndex++
 
-	// TODO: allocate correct size in BeginBlock instead of using append
```
Comment (ramilexe) [inline ethereum/pending.go:160]: It is correct todo. Issue for that https://github.com/tendermint/ethermint/issues/72

**Review by ramilexe — CHANGES_REQUESTED** [review]
Please return fix for issue #14

File: ethereum/pending.go (hunk 3 above)
Comment (adrianbrink) [reply ethereum/pending.go:160]: Yes, but I want to keep the todo's out of the codebase and on github instead.

**Review by ramilexe — APPROVED** [review]

File: ethereum/pending.go (hunk 3 above)
Comment (ramilexe) [reply ethereum/pending.go:160]: Ok, got it

**Pull request merged**